### PR TITLE
add tool to measure coverage

### PR DESCRIPTION
**Pull request by michaellilltokiwa**
this makes it easier to see which error message are not hit by the tests.
image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1079,3 +1085,18 @@ show_release_notes:
...
+$(JARS_JACOCO_AGENT):
+	mkdir -p $(@D)
+	curl $(JARS_JACOCO_AGENT_URL) --output $@
+	unzip $(JARS_JACOCO_AGENT) jacocoagent.jar -d $(BUILD_DIR)/jars
+
+$(JARS_JACOCO_CLI):
+	mkdir -p $(@D)
+	curl $(JARS_JACOCO_CLI_URL) --output $@
+
+coverage: $(FZ_INT) $(FZ_C) $(MOD_TERMINAL) $(MOD_JAVA_BASE) $(BUILD_DIR)/tests $(JARS_JACOCO_AGENT) $(JARS_JACOCO_CLI)
```
Comment (fridis) [inline Makefile:1098]: Please add a comment to explain what this does: running the tests both interpreted and with the C backend to collect coverage date using jacoco and then create html to show the results in a browser. 

In a later step, we might want to enable this via an option or env var given to `fz` such that we can easily collect coverage data.

File: Makefile [hunk 2]
```diff
@@ -1079,3 +1085,18 @@ show_release_notes:
...
+	(FUZION_JAVA_OPTIONS="-javaagent:../../jars/jacocoagent.jar=destfile=../../jacoco.exec" $(FZ_SRC)/bin/run_tests.sh $(BUILD_DIR) int) || true
```
Comment (fridis) [inline Makefile:1099]: The use of relative paths `../../jars/..` here is maybe a little brittle. Does `$(BUILD_DIR)/jars` work better?

**Review by fridis — CHANGES_REQUESTED** [review]
Does it make sense to use maven to get these jars? I do not have good memories of using maven, I was always confused about where code came from, but this might be my fault not knowing maven well enough.

File: Makefile (hunk 2 above)
Comment (michaellilltokiwa) [reply Makefile:1099]: @fridis `$(BUILD_DIR)/jars` is a relative path as well, I think? Maybe we could do somethink like `$(realpath $(BUILD_DIR)/jars)`. But is this less brittle? I don't know.

**Comment by michaellilltokiwa** [comment]
> Does it make sense to use maven to get these jars? I do not have good memories of using maven, I was always confused about where code came from, but this might be my fault not knowing maven well enough.

I don't know. An advantage we get by using maven is support by tooling to see vulnerable / outdated dependencies. Downside is one more dependency: `maven`.

**Pull request closed without merging**